test/tests/functional/pbs_basil_support.py

```diff
@@ -85,21 +90,25 @@ def setUp(self):
         if '$vnode_per_numa_node' in mom_config:
             momA.unset_mom_config('$vnode_per_numa_node', False)
 
-        # check if required BASIL version available on the machine.
-        xml_out = self.query_alps(self.basil_version[0], 'QUERY', 'ENGINE')
-        xml_tree = ET.parse(xml_out)
-        os.remove(xml_out)
+        momA.add_config({'$logevent': '0xffffffff'})
 
-        response = xml_tree.find(".//ResponseData")
-        status = response.attrib['status']
-
-        if status != "SUCCESS":
-            self.skipTest("Basil version %s not available on the platform" %
-                          (self.basil_version[0]))
+        # check if required BASIL version available on the machine.
+        for ver in self.basil_version:
+            xml_out = self.query_alps(ver, 'QUERY', 'ENGINE')
+            xml_tree = ET.parse(xml_out)
+            os.remove(xml_out)
+            response = xml_tree.find(".//ResponseData")
+            status = response.attrib['status']
+            if status == "SUCCESS":
+                self.available_version = ver
+                break
+        if self.available_version == "":
+            self.skipTest("No supported basil version found on the platform.")
 
         # Reset nodes
         self.reset_nodes(momA.shortname)
 
+    @classmethod
```

Comment:
Without decorator should work fine, but the codacy will report this as an issue.

Reply:
Codacy should not report this as an issue.
In fact, having this as a class method means you are making changes to the `TestBasilQuery` class instead of the python object with class `TestBasilQuery`. How PTL is handling this I don't know, but it's misleading to a future test writer looking at this code.